File: Solutions in Javascript/UglyNumber-II.js
/**
 * @author : Anmol Agrawal
 */
/**
 * @param {number} n
 * @return {number}
 */
var nthUglyNumber = function(n) {
    let uglys  = [];
    uglys.push(1);
    let i2 = 0;
    let i3 = 0;
    let i5 = 0;
    while(uglys.length < n){
        let next = Math.min(Math.min(uglys[i2] * 2, uglys[i3] * 3), uglys[i5] * 5);
        if(next === uglys[i2] * 2){
            i2++;
        }
        if(next === uglys[i3] * 3){
            i3++;
        }
        if(next === uglys[i5] * 5){
            i5++;
        }
        uglys.push(next);
    }
    return uglys[n - 1];
};

File: Solutions in Java/UglyNumber-II.java
/**
 * @author : Anmol Agrawal
 */
class Solution {
    public int nthUglyNumber(int n) {
        List<Integer> uglys  = new ArrayList<>();
        uglys.add(1);
        int i2 = 0;
        int i3 = 0;
        int i5 = 0;
        while(uglys.size() < n){
            int next = (int) Math.min(Math.min(uglys.get(i2) * 2, uglys.get(i3) * 3), uglys.get(i5)* 5);
            if(next == uglys.get(i2) * 2){
                i2++;
            }
            if(next == uglys.get(i3) * 3){
                i3++;
            }
            if(next == uglys.get(i5) * 5){
                i5++;
            }
            uglys.add(next);
        }
        return uglys.get(n-1);
    }
}
